src/napari_spatialdata/_viewer.py

```diff
@@ -225,16 +222,15 @@ def add_sdata_labels(self, sdata: SpatialData, key: str, selected_cs: str, multi
 
         affine = _get_transform(sdata.labels[original_name], selected_cs)
         rgb_labels, _ = _adjust_channels_order(element=sdata.labels[original_name])
+        adata = _get_metadata_adata(sdata, key)
 
         self.viewer.add_labels(
             rgb_labels,
             name=key,
             affine=affine,
             metadata={
                 "sdata": sdata,
-                "adata": sdata.table[sdata.table.obs[sdata.table.uns["spatialdata_attrs"]["region_key"]] == key]
-                if sdata.table
-                else None,
+                "adata": adata,
                 "region_key": sdata.table.uns["spatialdata_attrs"]["instance_key"] if sdata.table else None,
```

Comment:
in lines 169 and 214 we have "region_key" on the right hand side, here we have "instance_key", I think this is a bug.

Reply:
Could this `instance_key` be the cause of the bug with reversed indices? See my comment in the discussion page.